### ISSUE TITLE
TypeError: node.params is not iterable

### ISSUE THREAD

**Issue by rtritto** [typescript-eslint/typescript-eslint#3517]
I get `TypeError: node.params is not iterable` with `eslint` command using yarn v2 (berry).

**Repro**
1. `yarn init -y`
2. `yarn set version berry`
3. `yarn add react`
4. `yarn add -D @babel/core @babel/eslint-parser @babel/preset-env @babel/preset-react @babel/preset-typescript @typescript-eslint/eslint-plugin @typescript-eslint/parser babel-eslint eslint eslint-config-prettier eslint-config-react-app eslint-plugin-flowtype eslint-plugin-import eslint-plugin-jsx-a11y eslint-plugin-react eslint-plugin-react-hooks typescript`
5. Create files:
- `.eslintrc.js`
```JS
module.exports = {
  extends: ['react-app'],
  parser: '@babel/eslint-parser',
  parserOptions: {
    babelOptions: {
      presets: ['@babel/preset-react']
    }
  }
}
```
- `.babelrc.js`
```JS
module.exports = {
  presets: [
    '@babel/preset-env',
    '@babel/preset-react',
    '@babel/preset-typescript'
  ]
}
```
- test.d.ts
```TS
type PersonProps = {
  myFunc: () => void
}
```
6. `yarn eslint test.d.ts`

**Expected Result**
`eslint` command completed.

**Actual Result**

```
Oops! Something went wrong! :(

ESLint: 7.28.0

TypeError: node.params is not iterable
Occurred while linting .\test.d.ts:1
    at UnusedVarsVisitor.visitFunctionTypeSignature (.\.yarn\$$virtual\@typescript-eslint-eslint-plugin-virtual-17717ab341\0\cache\@typescript-eslint-eslint-plugin-npm-4.26.1-f48a1ca265-000a7ac316.zip\node_modules\@typescript-eslint\eslint-plugin\dist\util\collectUnusedVariables.js:164:34)
    at UnusedVarsVisitor.visit (.\.yarn\cache\@typescript-eslint-scope-manager-npm-4.26.1-bfd655174f-14cf14d2fc.zip\node_modules\@typescript-eslint\scope-manager\dist\referencer\VisitorBase.js:72:21)
    at UnusedVarsVisitor.visitChildren (.\.yarn\cache\@typescript-eslint-scope-manager-npm-4.26.1-bfd655174f-14cf14d2fc.zip\node_modules\@typescript-eslint\scope-manager\dist\referencer\VisitorBase.js:59:22)
    at UnusedVarsVisitor.visit (.\.yarn\cache\@typescript-eslint-scope-manager-npm-4.26.1-bfd655174f-14cf14d2fc.zip\node_modules\@typescript-eslint\scope-manager\dist\referencer\VisitorBase.js:77:14)
    at UnusedVarsVisitor.visitChildren (.\.yarn\cache\@typescript-eslint-scope-manager-npm-4.26.1-bfd655174f-14cf14d2fc.zip\node_modules\@typescript-eslint\scope-manager\dist\referencer\VisitorBase.js:59:22)
    at UnusedVarsVisitor.visit (.\.yarn\cache\@typescript-eslint-scope-manager-npm-4.26.1-bfd655174f-14cf14d2fc.zip\node_modules\@typescript-eslint\scope-manager\dist\referencer\VisitorBase.js:77:14)
    at UnusedVarsVisitor.visitChildren (.\.yarn\cache\@typescript-eslint-scope-manager-npm-4.26.1-bfd655174f-14cf14d2fc.zip\node_modules\@typescript-eslint\scope-manager\dist\referencer\VisitorBase.js:54:30)
    at UnusedVarsVisitor.visit (.\.yarn\cache\@typescript-eslint-scope-manager-npm-4.26.1-bfd655174f-14cf14d2fc.zip\node_modules\@typescript-eslint\scope-manager\dist\referencer\VisitorBase.js:77:14)
    at UnusedVarsVisitor.visitChildren (.\.yarn\cache\@typescript-eslint-scope-manager-npm-4.26.1-bfd655174f-14cf14d2fc.zip\node_modules\@typescript-eslint\scope-manager\dist\referencer\VisitorBase.js:59:22)
    at UnusedVarsVisitor.visit (.\.yarn\cache\@typescript-eslint-scope-manager-npm-4.26.1-bfd655174f-14cf14d2fc.zip\node_modules\@typescript-eslint\scope-manager\dist\referencer\VisitorBase.js:77:14)
```

**Additional Info**

Output of `yarn eslint test.d.ts` command.
<details>
  <summary>Debug Info</summary>

```
  eslint:cli CLI args: [ 'test.d.ts', '--debug' ] +0ms
  eslint:cli Running on files +9ms
  eslintrc:config-array-factory Loading JSON config file: .\package.json +0ms
  eslintrc:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/**/node_modules/*' ], basePath: '.', loose: false } ] +0ms
  eslintrc:ignore-pattern   processed: { basePath: '.', patterns: [ '/**/node_modules/*' ] } +10ms
  eslintrc:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/**/node_modules/*' ], basePath: '.', loose: false } ] +2ms
  eslintrc:ignore-pattern   processed: { basePath: '.', patterns: [ '/**/node_modules/*' ] } +2ms
  eslint:file-enumerator Start to iterate files: [ 'test.d.ts' ] +0ms
  eslint:file-enumerator File: .\test.d.ts +1ms
  eslintrc:cascading-config-array-factory Load config files for .. +0ms
  eslintrc:cascading-config-array-factory No cache found: .. +1ms
  eslintrc:config-array-factory Loading JS config file: .\.eslintrc.js +36ms
  eslintrc:config-array-factory Config file found: .\.eslintrc.js +12ms
  eslintrc:config-array-factory Loading {extends:"react-app"} relative to .\.eslintrc.js +1ms
  eslintrc:config-array-factory Loaded: eslint-config-react-app@6.0.0 (.\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\index.js) +5ms
  eslintrc:config-array-factory Loading JS config file: .\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\index.js +2ms
  eslintrc:config-array-factory Loading {extends:".\\.yarn\\$$virtual\\eslint-config-react-app-virtual-c37f2e07c7\\0\\cache\\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\\node_modules\\eslint-config-react-app\\base.js"} relative to .\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\index.js +12ms
  eslintrc:config-array-factory package.json was not found: Qualified path resolution failed - none of the candidates can be found on the disk.

Source path: .\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\base.js\package.json
Rejected candidate: .\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\base.js\package.json
Rejected candidate: .\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\base.js\package.json.js
Rejected candidate: .\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\base.js\package.json.json
Rejected candidate: .\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\base.js\package.json.node

Require stack:
- .\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\index.js +8ms
  eslintrc:config-array-factory Loaded: .\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\base.js (.\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\base.js) +1ms
  eslintrc:config-array-factory Loading JS config file: .\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\base.js +3ms
  eslintrc:config-array-factory Loading parser "babel-eslint" from .\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\base.js +3ms
  eslintrc:config-array-factory Loaded: babel-eslint@10.1.0 (.\.yarn\$$virtual\babel-eslint-virtual-0e4ee21c9a\0\cache\babel-eslint-npm-10.1.0-6a6d2b1533-c872bb9476.zip\node_modules\babel-eslint\lib\index.js) +5ms
  eslintrc:config-array-factory Loading plugin "react" from .\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\base.js +6ms
  eslintrc:config-array-factory Loaded: eslint-plugin-react@7.24.0 (.\.yarn\$$virtual\eslint-plugin-react-virtual-48aa0abf4d\0\cache\eslint-plugin-react-npm-7.24.0-16621467df-256e25dc8d.zip\node_modules\eslint-plugin-react\index.js) +13ms
  eslintrc:config-array-factory Plugin .\.yarn\$$virtual\eslint-plugin-react-virtual-48aa0abf4d\0\cache\eslint-plugin-react-npm-7.24.0-16621467df-256e25dc8d.zip\node_modules\eslint-plugin-react\index.js loaded in: 403ms +404ms
  eslintrc:config-array-factory Loading plugin "import" from .\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\index.js +1ms
  eslintrc:config-array-factory Loaded: eslint-plugin-import@2.23.4 (.\.yarn\$$virtual\eslint-plugin-import-virtual-a86124ddc5\0\cache\eslint-plugin-import-npm-2.23.4-b5f1dc1c73-d15a470088.zip\node_modules\eslint-plugin-import\lib\index.js) +12ms
  eslintrc:config-array-factory Plugin .\.yarn\$$virtual\eslint-plugin-import-virtual-a86124ddc5\0\cache\eslint-plugin-import-npm-2.23.4-b5f1dc1c73-d15a470088.zip\node_modules\eslint-plugin-import\lib\index.js loaded in: 185ms +185ms
  eslintrc:config-array-factory Loading plugin "flowtype" from .\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\index.js +1ms
  eslintrc:config-array-factory Loaded: eslint-plugin-flowtype@5.7.2 (.\.yarn\$$virtual\eslint-plugin-flowtype-virtual-86a5603db8\0\cache\eslint-plugin-flowtype-npm-5.7.2-8ce183496a-7a522a6278.zip\node_modules\eslint-plugin-flowtype\dist\index.js) +10ms
  eslintrc:config-array-factory Plugin .\.yarn\$$virtual\eslint-plugin-flowtype-virtual-86a5603db8\0\cache\eslint-plugin-flowtype-npm-5.7.2-8ce183496a-7a522a6278.zip\node_modules\eslint-plugin-flowtype\dist\index.js loaded in: 140ms +141ms
  eslintrc:config-array-factory Loading plugin "jsx-a11y" from .\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\index.js +0ms
  eslintrc:config-array-factory Loaded: eslint-plugin-jsx-a11y@6.4.1 (.\.yarn\$$virtual\eslint-plugin-jsx-a11y-virtual-c4e756ca37\0\cache\eslint-plugin-jsx-a11y-npm-6.4.1-c0ffc3303a-680d13f5e3.zip\node_modules\eslint-plugin-jsx-a11y\lib\index.js) +15ms
  eslintrc:config-array-factory Plugin .\.yarn\$$virtual\eslint-plugin-jsx-a11y-virtual-c4e756ca37\0\cache\eslint-plugin-jsx-a11y-npm-6.4.1-c0ffc3303a-680d13f5e3.zip\node_modules\eslint-plugin-jsx-a11y\lib\index.js loaded in: 803ms +809ms
  eslintrc:config-array-factory Loading plugin "react-hooks" from .\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\index.js +1ms
  eslintrc:config-array-factory Loaded: eslint-plugin-react-hooks@4.2.0 (.\.yarn\$$virtual\eslint-plugin-react-hooks-virtual-70f777291e\0\cache\eslint-plugin-react-hooks-npm-4.2.0-e09b1ea4ce-5378d16b5a.zip\node_modules\eslint-plugin-react-hooks\index.js) +9ms
  eslintrc:config-array-factory Plugin .\.yarn\$$virtual\eslint-plugin-react-hooks-virtual-70f777291e\0\cache\eslint-plugin-react-hooks-npm-4.2.0-e09b1ea4ce-5378d16b5a.zip\node_modules\eslint-plugin-react-hooks\index.js loaded in: 4ms +7ms
  eslintrc:config-array-factory Loading parser "@typescript-eslint/parser" from .\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\index.js +5ms
  eslintrc:config-array-factory Loaded: @typescript-eslint/parser@4.26.1 (.\.yarn\$$virtual\@typescript-eslint-parser-virtual-7f8845d5e7\0\cache\@typescript-eslint-parser-npm-4.26.1-e1f1ca8ed4-e8024ef340.zip\node_modules\@typescript-eslint\parser\dist\index.js) +4ms
  eslintrc:config-array-factory Loading plugin "@typescript-eslint" from .\.yarn\$$virtual\eslint-config-react-app-virtual-c37f2e07c7\0\cache\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\node_modules\eslint-config-react-app\index.js +636ms
  eslintrc:config-array-factory Loaded: @typescript-eslint/eslint-plugin@4.26.1 (.\.yarn\$$virtual\@typescript-eslint-eslint-plugin-virtual-17717ab341\0\cache\@typescript-eslint-eslint-plugin-npm-4.26.1-f48a1ca265-000a7ac316.zip\node_modules\@typescript-eslint\eslint-plugin\dist\index.js) +14ms
  eslintrc:config-array-factory Plugin .\.yarn\$$virtual\@typescript-eslint-eslint-plugin-virtual-17717ab341\0\cache\@typescript-eslint-eslint-plugin-npm-4.26.1-f48a1ca265-000a7ac316.zip\node_modules\@typescript-eslint\eslint-plugin\dist\index.js loaded in: 462ms +462ms
  eslintrc:config-array-factory Loading parser "@babel/eslint-parser" from .\.eslintrc.js +1ms
  eslintrc:config-array-factory Loaded: @babel/eslint-parser@7.14.5 (.\.yarn\$$virtual\@babel-eslint-parser-virtual-7bc994085f\0\cache\@babel-eslint-parser-npm-7.14.5-f9a360f1f5-2f24858075.zip\node_modules\@babel\eslint-parser\lib\index.cjs) +9ms
  eslintrc:cascading-config-array-factory Stop traversing because of 'root:true'. +3s
  eslint:rules Loading rule 'array-callback-return' (remaining=283) +0ms
  eslint:rules Loading rule 'default-case' (remaining=282) +6ms
  eslint:rules Loading rule 'dot-location' (remaining=281) +8ms
  eslint:rules Loading rule 'eqeqeq' (remaining=280) +4ms
  eslint:rules Loading rule 'new-parens' (remaining=279) +9ms
  eslint:rules Loading rule 'no-array-constructor' (remaining=278) +6ms
  eslint:rules Loading rule 'no-caller' (remaining=277) +3ms
  eslint:rules Loading rule 'no-cond-assign' (remaining=276) +8ms
  eslint:rules Loading rule 'no-const-assign' (remaining=275) +3ms
  eslint:rules Loading rule 'no-control-regex' (remaining=274) +2ms
  eslint:rules Loading rule 'no-delete-var' (remaining=273) +3ms
  eslint:rules Loading rule 'no-dupe-args' (remaining=272) +8ms
  eslint:rules Loading rule 'no-dupe-class-members' (remaining=271) +3ms
  eslint:rules Loading rule 'no-dupe-keys' (remaining=270) +1ms
  eslint:rules Loading rule 'no-duplicate-case' (remaining=269) +3ms
  eslint:rules Loading rule 'no-empty-character-class' (remaining=268) +9ms
  eslint:rules Loading rule 'no-empty-pattern' (remaining=267) +1ms
  eslint:rules Loading rule 'no-eval' (remaining=266) +2ms
  eslint:rules Loading rule 'no-ex-assign' (remaining=265) +4ms
  eslint:rules Loading rule 'no-extend-native' (remaining=264) +8ms
  eslint:rules Loading rule 'no-extra-bind' (remaining=263) +7ms
  eslint:rules Loading rule 'no-extra-label' (remaining=262) +8ms
  eslint:rules Loading rule 'no-fallthrough' (remaining=261) +4ms
  eslint:rules Loading rule 'no-func-assign' (remaining=260) +3ms
  eslint:rules Loading rule 'no-implied-eval' (remaining=259) +10ms
  eslint:rules Loading rule 'no-invalid-regexp' (remaining=258) +3ms
  eslint:rules Loading rule 'no-iterator' (remaining=257) +3ms
  eslint:rules Loading rule 'no-label-var' (remaining=256) +8ms
  eslint:rules Loading rule 'no-labels' (remaining=255) +3ms
  eslint:rules Loading rule 'no-lone-blocks' (remaining=254) +4ms
  eslint:rules Loading rule 'no-loop-func' (remaining=253) +7ms
  eslint:rules Loading rule 'no-mixed-operators' (remaining=252) +1ms
  eslint:rules Loading rule 'no-multi-str' (remaining=251) +9ms
  eslint:rules Loading rule 'no-native-reassign' (remaining=250) +6ms
  eslint:rules Loading rule 'no-negated-in-lhs' (remaining=249) +2ms
  eslint:rules Loading rule 'no-new-func' (remaining=248) +1ms
  eslint:rules Loading rule 'no-new-object' (remaining=247) +4ms
  eslint:rules Loading rule 'no-new-symbol' (remaining=246) +17ms
  eslint:rules Loading rule 'no-new-wrappers' (remaining=245) +2ms
  eslint:rules Loading rule 'no-obj-calls' (remaining=244) +1ms
  eslint:rules Loading rule 'no-octal' (remaining=243) +2ms
  eslint:rules Loading rule 'no-octal-escape' (remaining=242) +1ms
  eslint:rules Loading rule 'no-redeclare' (remaining=241) +1ms
  eslint:rules Loading rule 'no-regex-spaces' (remaining=240) +2ms
  eslint:rules Loading rule 'no-restricted-syntax' (remaining=239) +2ms
  eslint:rules Loading rule 'no-script-url' (remaining=238) +6ms
  eslint:rules Loading rule 'no-self-assign' (remaining=237) +6ms
  eslint:rules Loading rule 'no-self-compare' (remaining=236) +2ms
  eslint:rules Loading rule 'no-sequences' (remaining=235) +2ms
  eslint:rules Loading rule 'no-shadow-restricted-names' (remaining=234) +3ms
  eslint:rules Loading rule 'no-sparse-arrays' (remaining=233) +8ms
  eslint:rules Loading rule 'no-template-curly-in-string' (remaining=232) +2ms
  eslint:rules Loading rule 'no-this-before-super' (remaining=231) +1ms
  eslint:rules Loading rule 'no-throw-literal' (remaining=230) +2ms
  eslint:rules Loading rule 'no-undef' (remaining=229) +1ms
  eslint:rules Loading rule 'no-restricted-globals' (remaining=228) +4ms
  eslint:rules Loading rule 'no-unreachable' (remaining=227) +9ms
  eslint:rules Loading rule 'no-unused-expressions' (remaining=226) +1ms
  eslint:rules Loading rule 'no-unused-labels' (remaining=225) +2ms
  eslint:rules Loading rule 'no-unused-vars' (remaining=224) +2ms
  eslint:rules Loading rule 'no-use-before-define' (remaining=223) +7ms
  eslint:rules Loading rule 'no-useless-computed-key' (remaining=222) +6ms
  eslint:rules Loading rule 'no-useless-concat' (remaining=221) +3ms
  eslint:rules Loading rule 'no-useless-constructor' (remaining=220) +9ms
  eslint:rules Loading rule 'no-useless-escape' (remaining=219) +2ms
  eslint:rules Loading rule 'no-useless-rename' (remaining=218) +2ms
  eslint:rules Loading rule 'no-with' (remaining=217) +2ms
  eslint:rules Loading rule 'no-whitespace-before-property' (remaining=216) +1ms
  eslint:rules Loading rule 'require-yield' (remaining=215) +5ms
  eslint:rules Loading rule 'rest-spread-spacing' (remaining=214) +1ms
  eslint:rules Loading rule 'strict' (remaining=213) +2ms
  eslint:rules Loading rule 'unicode-bom' (remaining=212) +2ms
  eslint:rules Loading rule 'use-isnan' (remaining=211) +1ms
  eslint:rules Loading rule 'valid-typeof' (remaining=210) +2ms
  eslint:rules Loading rule 'no-restricted-properties' (remaining=209) +2ms
  eslint:rules Loading rule 'getter-return' (remaining=208) +11ms
  eslintrc:cascading-config-array-factory Configuration was determined: ConfigArray(5) [ { type: 'config', name: 'DefaultIgnorePattern', filePath: '', criteria: null, env: undefined, globals: undefined, ignorePattern: IgnorePattern { patterns: [Array], basePath: '.', loose: false }, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined }, { type: 'config', name: '.eslintrc.js » eslint-config-react-app » .\\.yarn\\$$virtual\\eslint-config-react-app-virtual-c37f2e07c7\\0\\cache\\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\\node_modules\\eslint-config-react-app\\base.js', filePath: '.\\.yarn\\$$virtual\\eslint-config-react-app-virtual-c37f2e07c7\\0\\cache\\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\\node_modules\\eslint-config-react-app\\base.js', criteria: null, env: { browser: true, commonjs: true, es6: true, jest: true, node: true }, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: { error: null, filePath: '.\\.yarn\\$$virtual\\babel-eslint-virtual-0e4ee21c9a\\0\\cache\\babel-eslint-npm-10.1.0-6a6d2b1533-c872bb9476.zip\\node_modules\\babel-eslint\\lib\\index.js', id: 'babel-eslint', importerName: '.eslintrc.js » eslint-config-react-app » .\\.yarn\\$$virtual\\eslint-config-react-app-virtual-c37f2e07c7\\0\\cache\\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\\node_modules\\eslint-config-react-app\\base.js', importerPath: '.\\.yarn\\$$virtual\\eslint-config-react-app-virtual-c37f2e07c7\\0\\cache\\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\\node_modules\\eslint-config-react-app\\base.js' }, parserOptions: { ecmaVersion: 2018, sourceType: 'module', ecmaFeatures: [Object] }, plugins: { react: [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: true, rules: { 'react/jsx-uses-vars': 'warn', 'react/jsx-uses-react': 'warn' }, settings: { react: [Object] } }, { type: 'config', name: '.eslintrc.js » eslint-config-react-app', filePath: '.\\.yarn\\$$virtual\\eslint-config-react-app-virtual-c37f2e07c7\\0\\cache\\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\\node_modules\\eslint-config-react-app\\index.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: undefined, parserOptions: undefined, plugins: { import: [Object], flowtype: [Object], 'jsx-a11y': [Object], 'react-hooks': [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'array-callback-return': 'warn', 'default-case': [Array], 'dot-location': [Array], eqeqeq: [Array], 'new-parens': 'warn', 'no-array-constructor': 'warn', 'no-caller': 'warn', 'no-cond-assign': [Array], 'no-const-assign': 'warn', 'no-control-regex': 'warn', 'no-delete-var': 'warn', 'no-dupe-args': 'warn', 'no-dupe-class-members': 'warn', 'no-dupe-keys': 'warn', 'no-duplicate-case': 'warn', 'no-empty-character-class': 'warn', 'no-empty-pattern': 'warn', 'no-eval': 'warn', 'no-ex-assign': 'warn', 'no-extend-native': 'warn', 'no-extra-bind': 'warn', 'no-extra-label': 'warn', 'no-fallthrough': 'warn', 'no-func-assign': 'warn', 'no-implied-eval': 'warn', 'no-invalid-regexp': 'warn', 'no-iterator': 'warn', 'no-label-var': 'warn', 'no-labels': [Array], 'no-lone-blocks': 'warn', 'no-loop-func': 'warn', 'no-mixed-operators': [Array], 'no-multi-str': 'warn', 'no-native-reassign': 'warn', 'no-negated-in-lhs': 'warn', 'no-new-func': 'warn', 'no-new-object': 'warn', 'no-new-symbol': 'warn', 'no-new-wrappers': 'warn', 'no-obj-calls': 'warn', 'no-octal': 'warn', 'no-octal-escape': 'warn', 'no-redeclare': 'warn', 'no-regex-spaces': 'warn', 'no-restricted-syntax': [Array], 'no-script-url': 'warn', 'no-self-assign': 'warn', 'no-self-compare': 'warn', 'no-sequences': 'warn', 'no-shadow-restricted-names': 'warn', 'no-sparse-arrays': 'warn', 'no-template-curly-in-string': 'warn', 'no-this-before-super': 'warn', 'no-throw-literal': 'warn', 'no-undef': 'error', 'no-restricted-globals': [Array], 'no-unreachable': 'warn', 'no-unused-expressions': [Array], 'no-unused-labels': 'warn', 'no-unused-vars': [Array], 'no-use-before-define': [Array], 'no-useless-computed-key': 'warn', 'no-useless-concat': 'warn', 'no-useless-constructor': 'warn', 'no-useless-escape': 'warn', 'no-useless-rename': [Array], 'no-with': 'warn', 'no-whitespace-before-property': 'warn', 'react-hooks/exhaustive-deps': 'warn', 'require-yield': 'warn', 'rest-spread-spacing': [Array], strict: [Array], 'unicode-bom': [Array], 'use-isnan': 'warn', 'valid-typeof': 'warn', 'no-restricted-properties': [Array], 'getter-return': 'warn', 'import/first': 'error', 'import/no-amd': 'error', 'import/no-anonymous-default-export': 'warn', 'import/no-webpack-loader-syntax': 'error', 'react/forbid-foreign-prop-types': [Array], 'react/jsx-no-comment-textnodes': 'warn', 'react/jsx-no-duplicate-props': 'warn', 'react/jsx-no-target-blank': 'warn', 'react/jsx-no-undef': 'error', 'react/jsx-pascal-case': [Array], 'react/no-danger-with-children': 'warn', 'react/no-direct-mutation-state': 'warn', 'react/no-is-mounted': 'warn', 'react/no-typos': 'error', 'react/require-render-return': 'error', 'react/style-prop-object': 'warn', 'jsx-a11y/alt-text': 'warn', 'jsx-a11y/anchor-has-content': 'warn', 'jsx-a11y/anchor-is-valid': [Array], 'jsx-a11y/aria-activedescendant-has-tabindex': 'warn', 'jsx-a11y/aria-props': 'warn', 'jsx-a11y/aria-proptypes': 'warn', 'jsx-a11y/aria-role': [Array], 'jsx-a11y/aria-unsupported-elements': 'warn', 'jsx-a11y/heading-has-content': 'warn', 'jsx-a11y/iframe-has-title': 'warn', 'jsx-a11y/img-redundant-alt': 'warn', 'jsx-a11y/no-access-key': 'warn', 'jsx-a11y/no-distracting-elements': 'warn', 'jsx-a11y/no-redundant-roles': 'warn', 'jsx-a11y/role-has-required-aria-props': 'warn', 'jsx-a11y/role-supports-aria-props': 'warn', 'jsx-a11y/scope': 'warn', 'react-hooks/rules-of-hooks': 'error', 'flowtype/define-flow-type': 'warn', 'flowtype/require-valid-file-annotation': 'warn', 'flowtype/use-flow-type': 'warn' }, settings: undefined }, { type: 'config', name: '.eslintrc.js » eslint-config-react-app#overrides[0]', filePath: '.\\.yarn\\$$virtual\\eslint-config-react-app-virtual-c37f2e07c7\\0\\cache\\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\\node_modules\\eslint-config-react-app\\index.js', criteria: { includes: [Array], excludes: null, basePath: '.' }, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: { error: null, filePath: '.\\.yarn\\$$virtual\\@typescript-eslint-parser-virtual-7f8845d5e7\\0\\cache\\@typescript-eslint-parser-npm-4.26.1-e1f1ca8ed4-e8024ef340.zip\\node_modules\\@typescript-eslint\\parser\\dist\\index.js', id: '@typescript-eslint/parser', importerName: '.eslintrc.js » eslint-config-react-app#overrides[0]', importerPath: '.\\.yarn\\$$virtual\\eslint-config-react-app-virtual-c37f2e07c7\\0\\cache\\eslint-config-react-app-npm-6.0.0-c5908e735c-ab983e9a31.zip\\node_modules\\eslint-config-react-app\\index.js' }, parserOptions: { ecmaVersion: 2018, sourceType: 'module', ecmaFeatures: [Object], warnOnUnsupportedTypeScriptVersion: true }, plugins: { '@typescript-eslint': [Object] }, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: { 'default-case': 'off', 'no-dupe-class-members': 'off', 'no-undef': 'off', '@typescript-eslint/consistent-type-assertions': 'warn', 'no-array-constructor': 'off', '@typescript-eslint/no-array-constructor': 'warn', 'no-redeclare': 'off', '@typescript-eslint/no-redeclare': 'warn', 'no-use-before-define': 'off', '@typescript-eslint/no-use-before-define': [Array], 'no-unused-expressions': 'off', '@typescript-eslint/no-unused-expressions': [Array], 'no-unused-vars': 'off', '@typescript-eslint/no-unused-vars': [Array], 'no-useless-constructor': 'off', '@typescript-eslint/no-useless-constructor': 'warn' }, settings: undefined }, { type: 'config', name: '.eslintrc.js', filePath: '.\\.eslintrc.js', criteria: null, env: undefined, globals: undefined, ignorePattern: undefined, noInlineConfig: undefined, parser: { error: null, filePath: '.\\.yarn\\$$virtual\\@babel-eslint-parser-virtual-7bc994085f\\0\\cache\\@babel-eslint-parser-npm-7.14.5-f9a360f1f5-2f24858075.zip\\node_modules\\@babel\\eslint-parser\\lib\\index.cjs', id: '@babel/eslint-parser', importerName: '.eslintrc.js', importerPath: '.\\.eslintrc.js' }, parserOptions: { babelOptions: [Object] }, plugins: undefined, processor: undefined, reportUnusedDisableDirectives: undefined, root: undefined, rules: undefined, settings: undefined } ] on . +356ms
  eslintrc:ignore-pattern Create with: [ IgnorePattern { patterns: [ '/**/node_modules/*' ], basePath: '.', loose: false } ] +4s
  eslintrc:ignore-pattern   processed: { basePath: '.', patterns: [ '/**/node_modules/*' ] } +2ms
  eslintrc:ignore-pattern Check {
  filePath: '.\\test.d.ts',
  dot: false,
  relativePath: 'test.d.ts',
  result: false
} +5ms
  eslint:cli-engine Lint .\test.d.ts +0ms
  eslint:linter Linting code for .\test.d.ts (pass 1) +0ms
  eslint:linter Verify +2ms
  eslint:linter With ConfigArray: .\test.d.ts +12ms
  eslint:linter An error occurred while traversing +568ms
  eslint:linter Filename: .\test.d.ts +1ms
  eslint:linter Line: 1 +4ms
  eslint:linter Parser Options: {
  ecmaFeatures: { globalReturn: false, jsx: true },
  ecmaVersion: 9,
  babelOptions: { presets: [ '@babel/preset-react' ] },
  sourceType: 'module',
  warnOnUnsupportedTypeScriptVersion: true
} +3ms
  eslint:linter Parser Path: .\.yarn\$$virtual\@babel-eslint-parser-virtual-7bc994085f\0\cache\@babel-eslint-parser-npm-7.14.5-f9a360f1f5-2f24858075.zip\node_modules\@babel\eslint-parser\lib\index.cjs +1ms
  eslint:linter Settings: { react: { version: 'detect' } } +1ms
```

</details>

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@babel/core` | `7.14.5` |
| `@babel/eslint-parser` | `7.14.5` |
| `@babel/preset-env` | `7.14.5` |
| `@babel/preset-react` | `7.14.5` |
| `@babel/preset-typescript` | `7.14.5` |
| `@typescript-eslint/eslint-plugin` | `4.26.1` |
| `@typescript-eslint/parser` | `4.26.1` |
| `babel-eslint` | `10.1.0` |
| `eslint` | `7.28.0` |
| `eslint-config-prettier` | `8.3.0` |
| `eslint-config-react-app` | `6.0.0` |
| `eslint-plugin-flowtype` | `5.7.2` |
| `eslint-plugin-import` | `2.23.4` |
| `eslint-plugin-jsx-a11y` | `6.4.1` |
| `eslint-plugin-react` | `7.24.0` |
| `eslint-plugin-react-hooks` | `4.2.0` |
| `react` | `17.0.2` |
| `typescript` | `4.3.2` |
| `node` | `14.17.0` |
| `yarn` | `2.4.2` |

**Comment by bradzacher** [comment]
> @babel/eslint-parser

There are known inconsistencies between the babel parser and our parser.
We recommend using our parser.

**Comment by rtritto** [comment]
Thanks!
I just replaced the parser with `@typescript-eslint/parser` in `.eslintrc`.
